### ISSUE TITLE
Issue with Filtering Null Dates in Power BI Using VEGA and DENEB

### ISSUE THREAD

**Issue by CSalcedoDataBI** [deneb-viz/deneb#446]
---

Some time ago, I was working with a date-type column in a Power BI model. I found that certain combinations in my model did not bring dates, only null values. To solve this, I tried applying filters in VEGA as follows:

```json
{
  "type": "filter",
  "expr": "datum.Date !== null"
}
```

However, this filter did not remove the null values. I tried changing the format to text, and in that case, it worked. I also tried the following expression:

```json
{
  "type": "filter",
  "expr": "datum.Date > null"
}
```

And this also worked. Additionally, when I tested these filters in the web editor, everything worked correctly, removing all the null values:

```json
{
  "type": "filter",
  "expr": "datum.Date !== null"
}
```

I even tried keeping only the null values, and it worked. However, when I tried to replicate this in DENEB, the filters did not work as expected. I suspect this is an issue that can be easily replicated with any Date or Datetime column in Power BI models. Therefore, I have not included a specific example, but I am attaching some images showing how it works on the web and in DENEB.

---
Deneb Editor:
![image](https://github.com/deneb-viz/deneb-viz.github.io/assets/73630372/cd587803-75f2-4cd9-874c-3f1edf9af995)
---
Web Editor:
![PuMtExcdX020240521-075948 Vega_Editor_y_1_página_más_-_ITM_Microsoft​_Edge](https://github.com/deneb-viz/deneb-viz.github.io/assets/73630372/c2b91af1-191b-487f-8c2f-445995ccd816)

https://vega.github.io/editor/#/gist/afe0b3e57cc62392a348260a46bca52c/Filtering Null Dates.json

**Comment by dm-p** [comment]
[transferring issue from doc site repo to visual repo - please try to create issues with Deneb in this repo rather than that one]

**Comment by CSalcedoDataBI** [comment]
Thank you, Daniel, for your prompt response and for testing this. In the following link, I have shared a sample PBIX file where you can observe the issue. I have also included a GIF showing the behavior of the error. You can download it here: [link to PBIX file](https://github.com/CSalcedoDataBI/SampleDataSets/raw/main/IssueWithFilteringNullDatesInPowerBIUsingVEGAandDENEB.pbix)

![drURA32lPE20240521-180439 Example](https://github.com/deneb-viz/deneb/assets/73630372/6e2bd7be-a369-42a8-bab0-d906080fd125)

**Comment by dm-p** [comment]
Thanks - this clarifies perfectly. It looks like a regression in the `datetime` data type handling, and this is getting injected as an object of type `Invalid Date` rather than `null` (and therefore not eligible for this condition).

This has been updated to ensure that `null` values for these field types are correctly stored as expected (and an opportunity to show off commenting😉):

https://github.com/deneb-viz/deneb/assets/10572054/24d7a510-3c2f-4f51-bcb2-c5d4da32e428

This fix is [included in the latest 1.7 alpha build](https://github.com/deneb-viz/deneb/releases/tag/alpha) for you to test and validate.

**Comment by CSalcedoDataBI** [comment]
Wow! I love the new look of Deneb 1.7. I’m going to play more with this version. I’ve tested the different filters, both to filter only null values and to discard them, and everything works perfectly. Many thanks, @Daniel Marsh-Patrick, for all your hard and excellent work; you can see it in the gif.
![PBIDesktop_reodmOV34N](https://github.com/deneb-viz/deneb/assets/73630372/7242a846-537b-4075-bb23-53105e320d8c)

**Comment by dm-p** [comment]
Glad it works for you! I'll mark this as closed, and when 1.7 gets published, the change will become GA.

Hope you enjoy the new features in 1.7!
